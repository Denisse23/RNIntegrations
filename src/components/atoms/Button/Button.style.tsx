import styled from '@emotion/native';
import { ThemeType } from '@theme/Themes';
import { ButtonProps } from './Button';

const getButtonBackgroundColor = (
  theme: ThemeType,
  type: ButtonProps['type'],
) => {
  switch (type) {
    case 'primary':
      return theme.colors.primary500;
    case 'secondary':
      return theme.colors.secondary500;
    case 'primaryText':
      return theme.colors.background100;
    case 'secondaryText':
      return theme.colors.background100;
  }
};

export const StyledButton = styled.TouchableOpacity<ButtonProps>(
  ({ theme, type }) => ({
    backgroundColor: getButtonBackgroundColor(theme, type),
    alignItems: 'center',
    padding: theme.spacings.xl,
    borderRadius: theme.borderRadius.lg,
    opacity: theme.opacities[300],
  }),
);
