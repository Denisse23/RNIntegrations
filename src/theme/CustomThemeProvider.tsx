import React from 'react';
import { useColorScheme } from 'react-native';
import { ThemeProvider } from '@emotion/react';
import { ColorScheme, DarkTheme, LightTheme } from './Themes';

const CustomThemeProvider = ({ children }: { children: React.ReactNode }) => {
  const scheme = useColorScheme();

  return (
    <ThemeProvider theme={scheme === ColorScheme.DARK ? DarkTheme : LightTheme}>
      {children}
    </ThemeProvider>
  );
};

export default CustomThemeProvider;
