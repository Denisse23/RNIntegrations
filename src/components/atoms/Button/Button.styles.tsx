import styled from '@emotion/native';
import { ThemeType } from '@theme/Themes';
import { ButtonProps } from './Button';
import { Shadow } from '@styles/SharedStyles';
import { scaleBySize } from '../../../utils/Scale';

const getButtonBackgroundColor = (
  theme: ThemeType,
  type: ButtonProps['type'],
) => {
  switch (type) {
    case 'primary':
      return theme.colors.background400;
    case 'secondary':
      return theme.colors.background300;
    case 'primaryText':
      return theme.colors.background100;
    case 'secondaryText':
      return theme.colors.background100;
  }
};

export const StyledButton = styled.TouchableOpacity<ButtonProps>(
  ({ theme, type }) => ({
    backgroundColor: getButtonBackgroundColor(theme, type),
    alignItems: 'center',
    justifyContent: 'center',
    height: scaleBySize(46),
    borderRadius: theme.borderRadius.lg,
    ...Shadow,
  }),
);
