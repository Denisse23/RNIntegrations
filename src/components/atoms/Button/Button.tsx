import React from 'react';
import { useTheme } from '@emotion/react';
import { TouchableOpacityProps } from 'react-native';
import { CustomText } from '@components/atoms';
import { StyledButton } from './Button.styles';

export interface ButtonProps extends TouchableOpacityProps {
  type: 'primary' | 'secondary' | 'primaryText' | 'secondaryText';
  text: string;
}

const Button = ({ type, text, testID, ...rest }: ButtonProps) => {
  const theme = useTheme();
  const buttonTextColor = () => {
    switch (type) {
      case 'primary':
        return theme.colors.text300;
      case 'secondary':
        return theme.colors.text100;
      case 'primaryText':
        return theme.colors.text300;
      case 'secondaryText':
        return theme.colors.text200;
    }
  };

  return (
    <StyledButton
      type={type}
      text={text}
      activeOpacity={theme.opacities[300]}
      testID={testID}
      accessibilityLabel={testID}
      {...rest}
    >
      <CustomText
        type={'bold'}
        size={'md'}
        style={{ color: buttonTextColor() }}
      >
        {text}
      </CustomText>
    </StyledButton>
  );
};

export default Button;
