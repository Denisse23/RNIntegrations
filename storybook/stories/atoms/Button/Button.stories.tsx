import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { Button } from '@components/atoms';
import { text, boolean } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import CustomThemeProvider from '@theme/CustomThemeProvider';
import CenterView from '../../CenterView';

storiesOf('Button', module)
  .addDecorator(story => <CenterView>{story()}</CenterView>)
  .add('primary', () => (
    <CustomThemeProvider>
      <Button
        disabled={boolean('Disabled', false)}
        text={text('Text', 'Primary')}
        onPress={action('onPressButton')}
        type="primary"
      />
    </CustomThemeProvider>
  ))
  .add('secondary', () => (
    <CustomThemeProvider>
      <Button
        disabled={boolean('Disabled', false)}
        text={text('Text', 'Secondary')}
        onPress={action('onPressButton')}
        type="secondary"
      />
    </CustomThemeProvider>
  ))
  .add('primaryText', () => (
    <CustomThemeProvider>
      <Button
        disabled={boolean('Disabled', false)}
        text={text('Text', 'Primary Text')}
        onPress={action('onPressButton')}
        type="primaryText"
      />
    </CustomThemeProvider>
  ))
  .add('secondaryText', () => (
    <CustomThemeProvider>
      <Button
        disabled={boolean('Disabled', false)}
        text={text('Text', 'Secondary Text')}
        onPress={action('onPressButton')}
        type="secondaryText"
      />
    </CustomThemeProvider>
  ));
