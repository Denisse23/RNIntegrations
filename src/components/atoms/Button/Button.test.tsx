import React from 'react';
import { fireEvent, waitFor } from '@testing-library/react-native';
import CustomThemeProvider from '@theme/CustomThemeProvider';
import renderer from 'react-test-renderer';
import Button, { ButtonProps } from './Button';

const buttonTestTypes = [
  {
    type: 'primary',
    text: 'Primary Button',
  },
  {
    type: 'primaryText',
    text: 'Primary Text Button',
  },
  {
    type: 'secondary',
    text: 'Secondary Button',
  },
] as ButtonProps[];

describe('Button', () => {
  buttonTestTypes[0].onPress = jest.fn();

  let testButtonType = (button: ButtonProps) => {
    it(`${button.text} should match the snapshot`, () => {
      const tree = renderer.create(
        <CustomThemeProvider>
          <Button {...button} />
        </CustomThemeProvider>,
      );
      expect(tree).toMatchSnapshot();
    });
  };

  buttonTestTypes.forEach(button => {
    testButtonType(button);
  });

  it('should call on press', async () => {
    const tree = renderer.create(
      <CustomThemeProvider>
        <Button {...buttonTestTypes[0]} />
      </CustomThemeProvider>,
    );

    const button = tree.root.findByProps(buttonTestTypes[0]);

    await waitFor(() => {
      fireEvent.press(button);
    });

    expect(buttonTestTypes[0].onPress).toBeCalledTimes(1);
  });
});
