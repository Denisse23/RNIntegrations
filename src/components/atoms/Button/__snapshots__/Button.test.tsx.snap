// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Primary Button should match the snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5BE59",
      "borderRadius": 75,
      "height": 107.8125,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    size="md"
    style={
      Array [
        Object {
          "color": "#353434",
          "fontSize": 32.8125,
          "fontWeight": "700",
        },
        Object {
          "color": "#353434",
        },
      ]
    }
    type="bold"
  >
    Primary Button
  </Text>
</View>
`;

exports[`Button Primary Text Button should match the snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FAFAFA",
      "borderRadius": 75,
      "height": 107.8125,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    size="md"
    style={
      Array [
        Object {
          "color": "#353434",
          "fontSize": 32.8125,
          "fontWeight": "700",
        },
        Object {
          "color": "#353434",
        },
      ]
    }
    type="bold"
  >
    Primary Text Button
  </Text>
</View>
`;

exports[`Button Secondary Button should match the snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#353434",
      "borderRadius": 75,
      "height": 107.8125,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    size="md"
    style={
      Array [
        Object {
          "color": "#353434",
          "fontSize": 32.8125,
          "fontWeight": "700",
        },
        Object {
          "color": "#FAFAFA",
        },
      ]
    }
    type="bold"
  >
    Secondary Button
  </Text>
</View>
`;

exports[`Button Secondary Text Button should match the snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FAFAFA",
      "borderRadius": 75,
      "height": 107.8125,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    size="md"
    style={
      Array [
        Object {
          "color": "#353434",
          "fontSize": 32.8125,
          "fontWeight": "700",
        },
        Object {
          "color": "#A9A8A3",
        },
      ]
    }
    type="bold"
  >
    Secondary Text Button
  </Text>
</View>
`;
